fast_barcode_scanner/lib/src/overlays/rect_of_interest/rect_of_interest.dart

```diff
@@ -0,0 +1,129 @@
+import 'dart:ui';
+
+import 'package:fast_barcode_scanner_platform_interface/fast_barcode_scanner_platform_interface.dart';
+
+import '../../corner_point_utils.dart';
+
+typedef CodeFilter = bool Function(Barcode code);
+
+/// A simple description of a rect of interest. The Rect will be located in the
+/// center of the screen and will fill the width minus horizontal padding.
+abstract class RectOfInterest {
+  const RectOfInterest();
+
+  Rect rect(Rect previewWidgetRect);
+
+  CodeFilter codeFilter({
+    required Size analysisSize,
+    required Size previewSize,
+  }) {
+    return (code) {
+      final cornerPoints = code.cornerPoints;
+      if (cornerPoints != null) {
+        for (final cornerPoint in cornerPoints) {
+          final scaledCornerOffset = scaleCodeCornerPoint(
+            cornerPoint: cornerPoint.toOffset(),
+            analysisImageSize: analysisSize,
+            widgetSize: previewSize,
+          );
+          if (!rect(previewSize.toRect()).contains(scaledCornerOffset)) {
+            // this code is not completely contained in the rect of interest
+            return false;
+          }
+        }
+        return true;
+      } else {
+        return false;
+      }
+    };
+  }
+
+  List<Barcode> filterCodes({
+    required List<Barcode> codes,
+    required Size analysisSize,
+    required Size previewSize,
+  }) {
+    return codes.where((code) {
+      final cornerPoints = code.cornerPoints;
+      if (cornerPoints != null) {
+        for (final cornerPoint in cornerPoints) {
+          final scaledCornerOffset = scaleCodeCornerPoint(
+            cornerPoint: cornerPoint.toOffset(),
+            analysisImageSize: analysisSize,
+            widgetSize: previewSize,
+          );
+          if (!rect(previewSize.toRect()).contains(scaledCornerOffset)) {
+            // this code is not completely contained in the rect of interest
+            return false;
+          }
+        }
+        return true;
+      } else {
+        return false;
+      }
+    }).toList();
+  }
+}
+
+class WideRectOfInterest extends RectOfInterest {
```

Comment:
This is not great for discoverability.
It would be nice to be able to do something like `RectOfInterest.wide()`.

Maybe also a rect of interest that creates itself dynamically according to the code types that are gonna be scanned?

Reply:
I like the discoverability suggestion. I'm adding that.

However, I think I'd rather leave the selection of square or rectangle to the user.